src/reader/metadata.rs

```diff
@@ -1,4 +1,28 @@
-use std::io::{Read, Seek, SeekFrom};
+//! Parse ORC file tail metadata structures from file.
+//!
+//! File tail structure:
+//!
+//! ------------------
+//! |    Metadata    |
+//! |                |
+//! ------------------
+//! |     Footer     |
+//! |                |
+//! ------------------
+//! |  Postscript  |X|
+//! ------------------
+//!
+//! Where X is last byte in file indicating
+//! Postscript length in bytes.
+//!
+//! Footer and Metadata lengths are encoded in Postscript.
+//! Postscript is never compressed, Footer and Metadata
+//! may be compressed depending Postscript config value.
+//!
+//! If they are compressed then their lengths indicate their
+//! compressed lengths.
```

Comment:
Where possible lets start adding docs

I prefer adding docs about the spec here since it removes the need for people to have to refer to the spec separately when trying to understand the code